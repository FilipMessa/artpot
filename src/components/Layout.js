// @flow
import * as React from 'react'
import media from 'styled-media-query'

import styled, { ThemeProvider } from 'styled-components'
import SEO from './SEO'
import Header from './Header'
import Navigation from './Navigation'

import theme from '../theme'
import GlobalStyle from '../theme/GlobalStyles'

type Props = {
  children: React.Node,
}

const Container = styled.div`
  display: grid;
  grid-template-columns: 1fr 4fr 1fr;
  grid-template-areas:
    'header header header'
    'navbar content leftSide';

  padding: ${({ theme }) => theme.layout.padding};

  ${media.lessThan('medium')`
    grid-template-columns: none;
    grid-template-rows: 1fr;

    grid-gap: 20px;

    grid-template-areas:
      'header'
      'navbar'
      'content';
    `}
`

const LeftSide = styled.div`
  grid-area: leftSide;
  ${media.lessThan('medium')`
    display: none;
  `}
`

const HeaderW = styled.div`
  grid-area: header;
`

const Navbar = styled.div`
  grid-area: navbar;
`

const Content = styled.div`
  grid-area: content;
`

const Layout = ({ children }: Props) => (
  <ThemeProvider theme={theme}>
    <>
      <SEO />
      <Container>
        <HeaderW>
          <Header />
        </HeaderW>
        <Navbar>
          <Navigation />
        </Navbar>

        <Content>
          <div>{children}</div>
        </Content>
        <LeftSide />
      </Container>
      <GlobalStyle />
    </>
  </ThemeProvider>
)

export default Layout
