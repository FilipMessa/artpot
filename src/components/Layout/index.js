// @flow
import * as React from 'react'
import Helmet from 'react-helmet'
import styled, { ThemeProvider } from 'styled-components'
import { StaticQuery, graphql } from 'gatsby'
import theme from '../../theme'
import GlobalStyle from '../styled/GlobalStyles'
import Navigation from '../Navigation/index'
import Header from '../Header'

const Container = styled.div`
  padding: 2em;
`

export const PureLayout = ({ children }: { children: React.Node }) => (
  <Container>
    <Header />
    <div style={{ marginTop: '2em' }}>
      <Navigation />
    </div>
    <div>{children}</div>
  </Container>
)

const Layout = ({ children }: { children: React.Node }) => (
  <StaticQuery
    query={graphql`
      query SiteTitleQuery {
        site {
          siteMetadata {
            title
          }
        }
      }
    `}
    render={data => (
      <ThemeProvider theme={theme}>
        <>
          <Helmet
            title={data.site.siteMetadata.title}
            meta={[
              { name: 'description', content: 'Sample' },
              { name: 'keywords', content: 'sample, something' },
            ]}
          >
            <html lang="en" />
          </Helmet>
          <PureLayout>{children}</PureLayout>
          <GlobalStyle />
        </>
      </ThemeProvider>
    )}
  />
)

export default Layout
