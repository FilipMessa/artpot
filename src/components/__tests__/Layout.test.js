// @flow
import React from 'react'
import { render } from 'react-testing-library'
import Layout from '../Layout'

describe('Layout', () => {
  test('render', () => {
    const { baseElement } = render(<Layout>Child</Layout>)
    expect(baseElement).toMatchSnapshot()
  })
})
