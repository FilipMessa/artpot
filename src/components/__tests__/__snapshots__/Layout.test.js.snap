// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout render 1`] = `
<body>
  <div>
    <div>
      Loading (StaticQuery)
    </div>
  </div>
</body>
`;
